packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -578,18 +580,47 @@ export const Table = <
     );
   }, [data, getSubRows]);
 
-  const [rowHasParent] = React.useState(() => new WeakSet<T>());
-
   const getSubRowsWithSubComponents = React.useCallback(
-    (originalRow: T) => {
-      if (!rowHasParent.has(originalRow)) {
-        rowHasParent.add(originalRow);
-        return [originalRow];
+    (originalRow: T, relativeIndex: number) => {
+      console.log('row', originalRow);
+      // If originalRow represents a sub-component, don't add any sub-rows to it
+      if (originalRow[iuiId as any]) {
+        return [];
+      }
+
+      // If originalRow represents the top-most level row, add itself as a sub-row that will represent a subComponent.
+      // Add a symbol as a key on this sub-row to distinguish it from the real row.
+      // This distinction is needed as the sub-row needs to have all fields of the row since react-table expects even sub-rows to be rows (Row<T>).
```

Comment:
I think this comment is in the wrong place. It should be a few lines further down.

Reply:
I have moved this down and added another comment here.